extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxRequestHandler.java

```diff
@@ -127,8 +129,10 @@ private void dispatch(RoutingContext routingContext, InputStream is, VertxOutput
                     dispatcher.getDispatcher(), vertxResponse, requestContext, executor);
             vertxRequest.setInputStream(is);
             try {
-                ResteasyContext.pushContext(SecurityContext.class, new QuarkusResteasySecurityContext(request, routingContext));
-                ResteasyContext.pushContext(RoutingContext.class, routingContext);
+                Map<Class<?>, Object> map = new HashMap<>();
+                map.put(SecurityContext.class, new QuarkusResteasySecurityContext(request, routingContext));
+                map.put(RoutingContext.class, routingContext);
+                ResteasyContext.pushContextDataMap(map);
```

Comment:
I think you should be using the variant that creates a new `Closeable` context : `addCloseableContextDataLevel` that you can use in a try-with-resource.

Reply:
What about async requests?

Reply:
Async requests need to capture this context anyway, since we've no idea what thread they'll be in when they get executed. This doesn't destroy anything, BTW. It just pops it from the current context stack and restore the previous context map.